torchmultimodal/utils/attention.py

```diff
@@ -27,14 +27,16 @@ def get_extended_attention_mask(attention_mask: Tensor) -> Tensor:
         extended_attention_mask (Tensor):
             The broadcastable attention mask, with the same dtype as ``attention_mask.dtype``.
     """
-    # We can provide a self-attention mask of dimensions [batch_size, query_seq_length, key_seq_length]
-    # ourselves in which case we just need to make it broadcastable to all heads,
-    # [batch_size, num_heads, query_seq_length, key_seq_length].
-    if attention_mask.dim() == 3:
+    if attention_mask.dim() == 4:
```

Comment:
Can we add comments to this codepath as well? 
Also update the docstring since we are now accepting masks with head dim.

```
Accepting two types of attention masks:
        - Causal: masks that prevent attending to future positions of dimensions
            ``(batch_size, query_seq_len, key_seq_len)``
        - Padding: masks that prevent attending to token paddings of dimensions
            ``(batch_size, seq_len)``
```

Reply:
Non blocking Q: this utility is ambiguous as it puts strong assumption around the original dims of the masks.
For example, dim=3 could mean both `(b, h, seq_len)` and `(b, q_seq_len, k_seq_len)`.
May introduce bugs unless we have a convention for the original dims. Better be careful with it.

Reply:
Yeah this is a fair point. I think we should not expect users to pass `(b, h, seq_len)` very often, since the inclusion of `h` implies they have already broadcasted (and if using this util the broadcasting will have already been to 4D). But agree there can be some ambiguity here. We could add a flag, but not sure that is the best approach either...

Reply:
nit: If the more common case is passing without either batch or head dims, perhaps update the docstring to reflect that expectation? Like:

```
Accepting two types of attention masks:
        - Causal: masks that prevent attending to future positions of dimensions
            ``(query_seq_len, key_seq_len)``
        - Padding: masks that prevent attending to token paddings of dimensions
            ``(seq_len)``
```

Reply:
Oh sorry I should clarify... I mean that the common cases are `(b, query_seq_len, key_seq_len)` or `(b, seq_len)`. So in either case we still have `b`, it is just `h` that is missing